### PR TITLE
feat(ecocredit): setup ecocredit for ORM integration

### PR DESCRIPTION
**Pull request by technicallyty**
## Description
- Beginning of ecocredit makeover to use ORM
- Includes some additions utilized in #732 

Ref: #696 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/regen/ecocredit/basket/v1/events.proto [hunk 1]
```diff
@@ -4,7 +4,7 @@ package regen.ecocredit.basket.v1;
 
 import "regen/ecocredit/basket/v1/types.proto";
 
-option go_package = "github.com/regen-network/regen-ledger/x/ecocredit/basket";
+option go_package = "github.com/regen-network/regen-ledger/x/ecocredit/basket/v1";
```
Comment (aaronc) [inline proto/regen/ecocredit/basket/v1/events.proto:7]: Do we want to change this now?

**Review by aaronc — APPROVED** [review]

File: proto/regen/ecocredit/basket/v1/events.proto (hunk 1 above)
Comment (technicallyty) [reply proto/regen/ecocredit/basket/v1/events.proto:7]: if we're going to have multiple versions of proto packages, i think so. 

we kept running into an issue with overwriting due to different versioned proto packages with the same go_option, i.e. v1alpha1 and v1alpha2 both pointed to `github.com/regen-network/regen-ledger/x/ecocredit`.

File: proto/regen/ecocredit/v1beta1/query.proto [hunk 2]
```diff
@@ -24,7 +24,14 @@ service Query {
         "/regen/ecocredit/v1beta1/classes/{class_id}";
   }
 
-  // Projects queries for all projects within a class with pagination.
+  // ClassIssuers queries for the addresses of the issuers for a credit class.
+  rpc ClassIssuers(QueryClassIssuersRequest) returns (QueryClassIssuersResponse) {
+    option (google.api.http).get =
+      "/regen/ecocredit/v1alpha2/classes/{class_id}/issuers";
```
Comment (ryanchristo) [inline proto/regen/ecocredit/v1beta1/query.proto:30]:
```suggestion
      "/regen/ecocredit/v1beta1/classes/{class_id}/issuers";
```

File: x/ecocredit/server/core/keeper.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+func NewKeeper(ss ecocreditv1beta1.StateStore, bk ecocredit.BankKeeper, params ParamKeeper) Keeper {
+	return Keeper{
+		stateStore: ss,
+		bankKeeper: bk,
+		params:     params,
+	}
+}
+
+// TODO: uncomment when impl
+//var _ v1beta1.MsgServer = &Keeper{}
```
Comment (ryanchristo) [inline x/ecocredit/server/core/keeper.go:29]: Are we tracking this?

**Review by ryanchristo — APPROVED** [review]
Looks good to me. One change needed. Pre-approving.

File: proto/regen/ecocredit/v1beta1/query.proto (hunk 2 above)
Comment (technicallyty) [reply proto/regen/ecocredit/v1beta1/query.proto:30]: 😳 😳 thanks for catching that!

File: x/ecocredit/server/core/keeper.go (hunk 3 above)
Comment (technicallyty) [reply x/ecocredit/server/core/keeper.go:29]: https://github.com/regen-network/regen-ledger/issues/696#issue-1111175005 added a line to this issue 😄

**Comment by technicallyty** [comment]
looks like the linter is still busted, should we disable it on master? @aaronc ?

**Comment by aaronc** [comment]
Disabled

**Pull request merged**